cmd/gitserver/server/server.go

```diff
@@ -1405,6 +1417,27 @@ func (s *Server) ensureRevision(ctx context.Context, repo api.RepoName, url, rev
 	return true
 }
 
+// quickSymbolicRefHead best-effort mimics the execution of `git symbolic-ref HEAD`, but doesn't exec a child process.
+// It just reads the .git/HEAD file from the bare git repository directory.
+func quickSymbolicRefHead(dir GitDir) (string, error) {
+	// See if HEAD contains a commit hash and fail if so.
+	head, err := ioutil.ReadFile(dir.Path("HEAD"))
+	if err != nil {
+		return "", err
+	}
+	head = bytes.TrimSpace(head)
+	if h := string(head); isAbsoluteRevision(h) {
+		return "", errors.New("ref HEAD is not a symbolic ref")
+	}
```

Comment:
```suggestion
	if isAbsoluteRevision(string(head)) {
		return "", errors.New("ref HEAD is not a symbolic ref")
	}
```

I think it's clearer without the temporary variable